src/main/java/randoop/generation/CoverageTracker.java

```diff
@@ -0,0 +1,172 @@
+import org.jacoco.core.analysis.Analyzer;
+import org.jacoco.core.analysis.CoverageBuilder;
+import org.jacoco.core.analysis.IClassCoverage;
+import org.jacoco.core.analysis.IMethodCoverage;
+import org.jacoco.core.data.ExecutionData;
+import org.jacoco.core.data.ExecutionDataReader;
+import org.jacoco.core.data.ExecutionDataStore;
+import org.jacoco.core.data.IExecutionDataVisitor;
+import org.jacoco.core.data.ISessionInfoVisitor;
+import org.jacoco.core.data.SessionInfo;
+import randoop.main.GenInputsAbstract;
+import randoop.types.ClassOrInterfaceType;
+
+/**
+ * Tracks the branch coverage of each method under test. Specifically, for each method under test,
+ * this class records the total number of branches and the number of branches that have not been
+ * covered in generated tests. This class periodically updates branch coverage information for each
+ * method from Jacoco's data structures.
+ */
+public class CoverageTracker {
+  /** In-memory store of the coverage information for all classes under test. */
+  private final ExecutionDataStore executionData = new ExecutionDataStore();
+
+  /**
+   * Map from method name to uncovered branch ratio. In cases where a method's total branches is
+   * zero, the missed ratio is NaN, and this map uses zero instead.
+   */
+  private final Map<String, Double> branchCoverageMap = new HashMap<>();
+
+  /** Names of all the classes under test */
+  private final Set<String> classesUnderTest = new HashSet<>();
+
+  /**
+   * Initialize the coverage tracker.
+   *
+   * @param classInterfaceTypes names of all the classes under test
+   */
+  public CoverageTracker(Set<ClassOrInterfaceType> classInterfaceTypes) {
+    for (ClassOrInterfaceType classOrInterfaceType : classInterfaceTypes) {
+      classesUnderTest.add(classOrInterfaceType.getRuntimeClass().getName());
+    }
+  }
+
+  /**
+   * Retrieve execution data from the Jacoco Java agent and merge the coverage information into
+   * {@code executionData}.
+   */
+  private void collectCoverageInformation() {
+    try {
+      // Retrieve the execution data from the Jacoco Java agent.
+      final InputStream execDataStream;
+      try {
+        execDataStream = new ByteArrayInputStream(RT.getAgent().getExecutionData(false));
+      } catch (IllegalStateException e) {
+        System.err.println(
+            "If the error notes: 'JaCoCo agent not started', the issue is likely "
+                + "that the Jacoco agent is not included as a Java agent.");
+        System.err.println(
+            "To do so, add "
+                + "'-Xbootclasspath/a:/path/to/jacocoagent.jar -javaagent:jacocoagent.jar' "
+                + "to the command line argument.");
+        throw (e);
+      }
+
+      final ExecutionDataReader reader = new ExecutionDataReader(execDataStream);
+
+      // The reader requires a session info visitor, however we don't need any information from it.
+      reader.setSessionInfoVisitor(
+          new ISessionInfoVisitor() {
```

Comment:
Minor:  I would find it clearer to create a private singleton class `DummySessionInfoVisitor` and use it here.